src/read.lisp

```diff
@@ -77,9 +77,9 @@
 (defun make-string-stream (string)
   (cons string 0))
 
-(defun %peek-char (stream)
-  (and (< (cdr stream) (length (car stream)))
-       (char (car stream) (cdr stream))))
+(defun %peek-char (stream &optional (look-ahead 0))
```

Comment:
We shouldn't use look-ahead. We can't support that for arbitrary streams.

The current in-memory string representation is just a basic case for convenience, but the reader shouldn't assume that it's working on it.

I will not include the reader changes yet.